History/HistoryEntry.h

```diff
@@ -61,9 +61,82 @@ struct HistoryInlineRef
     bool        mIsRemote   : 1;
     bool        mIsTag      : 1;
     bool        mIsStash    : 1;
+    bool        mIsDetached : 1;
+
+    bool operator ==(const HistoryInlineRef& other) const
+    {
+        if ( this == &other ) {
+            return true;
+        }
+
+        return  mRefName    == other.mRefName       &&
+                mIsBranch   == other.mIsBranch      &&
+                mIsCurrent  == other.mIsCurrent     &&
+                mIsRemote   == other.mIsRemote      &&
+                mIsTag      == other.mIsTag         &&
+                mIsStash    == other.mIsStash       &&
+                mIsDetached == other.mIsDetached    ;
+    }
+
+    inline bool operator !=(const HistoryInlineRef& other) const {
+        return !( *this == other );
+    }
+
+    inline bool detachedHEAD() const
+    {
+        return mIsDetached && !(mIsTag || mIsBranch || mIsStash);
+    }
+
+    inline bool nameLessThan(const HistoryInlineRef& other) const
+    {
+        return mRefName < other.mRefName;
+    }
+
+    bool operator <(const HistoryInlineRef& other) const
+    {
+        // sort order:
+        // - tag
+        // - detached HEAD
+        // - current branch
+        // - local branch
```

Comment:
>Thinking about this a bit more, I'd like to insert "Remote Tracking Branch" into the sort order between current branch and local branches - Would that make sense?

Yeah, that makes sense. Though it would look a bit weird. Actually this is worth another feature request to "get it right". See [here](https://redmine.macgitver.org/issues/238).

Btw.: Isn't it better to make the operator static? Or is it static by definition?